yt_dlp/extractor/tiktok.py

```diff
@@ -323,38 +337,32 @@ def _real_extract(self, url):
         raise ExtractorError('Video not available', video_id=video_id)
 
 
-class TikTokUserIE(InfoExtractor):
+class TikTokUserIE(TikTokBaseIE):
     IE_NAME = 'tiktok:user'
-    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\._]+)/?(?:$|[#?])'
+    _VALID_URL = r'https?://(?:www\.)?tiktok\.com/@(?P<id>[\w\.-]+)/?(?:$|[#?])'
     _TESTS = [{
         'url': 'https://tiktok.com/@corgibobaa?lang=en',
         'playlist_mincount': 45,
         'info_dict': {
             'id': '6935371178089399301',
         },
-        'skip': 'Cookies (not necessarily logged in) are needed.'
     }, {
         'url': 'https://www.tiktok.com/@meme',
         'playlist_mincount': 593,
         'info_dict': {
             'id': '79005827461758976',
         },
-        'skip': 'Cookies (not necessarily logged in) are needed.'
     }]
 
-    def _entries(self, url, user_id):
-        webpage = self._download_webpage(url, user_id)
-        own_id = self._search_regex(r'\"id\":\"(?P<userid>\d+)', webpage, user_id, default=None)
-        if not own_id:
-            raise ExtractorError('Cookies (not necessarily logged in) are needed.', expected=True)
-        secuid = self._search_regex(r'\"secUid\":\"(?P<secUid>[^\"]+)', webpage, user_id)
+    def _entries(self, webpage, user_id, username):  # TODO: Fix?
```

Comment:
what's with the `TODO`?

Reply:
https://github.com/yt-dlp/yt-dlp/issues/859

Reply:
is this PR not addressing that issue?

Reply:
(resolved on discord https://github.com/yt-dlp/yt-dlp/pull/925/commits/143ca2c32118e040d90deb3b972ca4ad32ff8b7e)